pkg/controller/operators/olm/operator.go

```diff
@@ -442,22 +460,37 @@ func (a *Operator) checkReplacementsAndUpdateStatus(csv *v1alpha1.ClusterService
 }
 
 func (a *Operator) updateInstallStatus(csv *v1alpha1.ClusterServiceVersion, installer install.StrategyInstaller, strategy install.Strategy, requeueConditionReason v1alpha1.ConditionReason) error {
-	installed, strategyErr := installer.CheckInstalled(strategy)
-	if installed {
+	apiServicesInstalled, apiServiceErr := a.areAPIServicesAvailable(csv.Spec.APIServiceDefinitions.Owned)
+	strategyInstalled, strategyErr := installer.CheckInstalled(strategy)
+	if strategyInstalled && apiServicesInstalled {
 		// if there's no error, we're successfully running
 		if csv.Status.Phase != v1alpha1.CSVPhaseSucceeded {
 			csv.SetPhase(v1alpha1.CSVPhaseSucceeded, v1alpha1.CSVReasonInstallSuccessful, "install strategy completed with no errors")
 		}
 		return nil
 	}
 
+	// TODO(Nick): check if apiServiceErr is unrecoverable
+
 	// installcheck determined we can't progress (e.g. deployment failed to come up in time)
 	if install.IsErrorUnrecoverable(strategyErr) {
 		csv.SetPhase(v1alpha1.CSVPhaseFailed, v1alpha1.CSVReasonInstallCheckFailed, fmt.Sprintf("install failed: %s", strategyErr))
 		return strategyErr
 	}
 
 	// if there's an error checking install that shouldn't fail the strategy, requeue with message
+	if apiServiceErr != nil {
+		csv.SetPhase(v1alpha1.CSVPhaseInstalling, requeueConditionReason, fmt.Sprintf("APIServices installing: %s", apiServiceErr))
+		a.requeueCSV(csv.GetName(), csv.GetNamespace())
```

Comment:
I'm not sure that this is the best way to ensure APIService GVK discovery is immediately detected. Watching APIServices only tells us when its status changes to available, but not when the expected GVKs are discoverable (which can happen much later). I expect this to swamp the CSV workqueue. 

@ecordell Could we potentially watch Discovery for changes or start a go-routine when we detect a change to an APIService that requeues the CSV when the appropriate GVKs become available?

Reply:
[ALM-749](https://jira.coreos.com/browse/ALM-749)